l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -2,133 +2,87 @@
     /// @dev Initialize the implementation to prevent Parity hack.
-    constructor(IZkSync _zkSync) reentrancyGuardInitializer {
-        zkSync = _zkSync;
+    constructor(IL1SharedBridge _sharedBridge) reentrancyGuardInitializer {
+        sharedBridge = _sharedBridge;
     }
 
-    /// @dev Initializes a contract bridge for later use. Expected to be used in the proxy
-    /// @dev During initialization deploys L2 bridge counterpart as well as provides some factory deps for it
-    /// @param _factoryDeps A list of raw bytecodes that are needed for deployment of the L2 bridge
-    /// @notice _factoryDeps[0] == a raw bytecode of L2 bridge implementation
-    /// @notice _factoryDeps[1] == a raw bytecode of proxy that is used as L2 bridge
-    /// @notice _factoryDeps[2] == a raw bytecode of token proxy
-    /// @param _l2TokenBeacon Pre-calculated address of the L2 token upgradeable beacon
-    /// @notice At the time of the function call, it is not yet deployed in L2, but knowledge of its address
-    /// @notice is necessary for determining L2 token address by L1 address, see `l2TokenAddress(address)` function
-    /// @param _governor Address which can change L2 token implementation and upgrade the bridge
-    /// @param _deployBridgeImplementationFee How much of the sent value should be allocated to deploying the L2 bridge
-    /// implementation
-    /// @param _deployBridgeProxyFee How much of the sent value should be allocated to deploying the L2 bridge proxy
-    function initialize(
-        bytes[] calldata _factoryDeps,
-        address _l2TokenBeacon,
-        address _governor,
-        uint256 _deployBridgeImplementationFee,
-        uint256 _deployBridgeProxyFee
-    ) external payable reentrancyGuardInitializer {
-        require(_l2TokenBeacon != address(0), "nf");
-        require(_governor != address(0), "nh");
-        // We are expecting to see the exact three bytecodes that are needed to initialize the bridge
-        require(_factoryDeps.length == 3, "mk");
-        // The caller miscalculated deploy transactions fees
-        require(msg.value == _deployBridgeImplementationFee + _deployBridgeProxyFee, "fee");
-        l2TokenProxyBytecodeHash = L2ContractHelper.hashL2Bytecode(_factoryDeps[2]);
-        l2TokenBeacon = _l2TokenBeacon;
+    /// @dev Initializes the reentrancy guard. Expected to be used in the proxy.
+    function initialize() external reentrancyGuardInitializer {}
 
-        bytes32 l2BridgeImplementationBytecodeHash = L2ContractHelper.hashL2Bytecode(_factoryDeps[0]);
-        bytes32 l2BridgeProxyBytecodeHash = L2ContractHelper.hashL2Bytecode(_factoryDeps[1]);
+    /// @dev transfer token to shared bridge as part of upgrade
+    function tranferTokenToSharedBridge(address _token, uint256 _amount) external {
+        require(msg.sender == address(sharedBridge), "Not shared bridge");
+        uint256 amount = IERC20(_token).balanceOf(address(this));
+        require(amount == _amount, "Incorrect amount");
```

Comment:
What if someone frontrun the transaction of `tranferTokenToSharedBridge`? Better to transfer the whole balance at once.